### PR TITLE
LPS-170580 | Add Autom. test ClayAlertToast#CanTransitionToFillGapOfDeletedToast

### PR DESCRIPTION
**Pull request by deborasoliveira**
**ClayAlertToast#CanTransitionToFillGapOfDeletedToast**
**Test Plan**

- Given Clay Sample Portlet
- And multiple toast alerts
- When an alert in the middle of the stack is closed
- Then the alerts transition to fill the stack

https://issues.liferay.com/browse/LPS-170580

### CODE REVIEW

**Comment by deborasoliveira** [comment]
ci:test:frontend

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayAlertToast.testcase [hunk 1]
```diff
@@ -72,6 +72,49 @@ definition {
...
+			var position_y = "710";
+			var positionAlertX = ClayAlertToast.getElementRectanglePositionX(
+				code = "#ToastAlertContainer .alert-success",
+				index = "0");
+			var positionAlertY = ClayAlertToast.getElementRectanglePositionY(
+				code = "#ToastAlertContainer .alert-success",
+				index = "0");
+
+			if ("${positionAlertX}" != "${position_x}") {
+				fail("FAIL. X coordinate positions do not match.");
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayAlertToast.testcase:109]: add a takeScreenshot(); right before the fail

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayAlertToast.testcase [hunk 2]
```diff
@@ -72,6 +72,49 @@ definition {
...
+			}
+
+			if ("${positionAlertY}" != "${position_y}") {
+				fail("FAIL. Y coordinate positions do not match.");
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayAlertToast.testcase:113]: add a takeScreenshot(); right before the fail

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayAlertToast.testcase [hunk 3]
```diff
@@ -72,6 +72,49 @@ definition {
 		}
 	}
 
+	@description = "LPS-170580 - Verify alert in the middle of stack is closed, the others fill the stack"
+	@priority = "4"
+	test CanTransitionToFillGapOfDeletedToast {
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayAlertToast.testcase:77]: Given the known first (top) alert position from two stacked toast alert success messages

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayAlertToast.testcase [hunk 4]
```diff
@@ -72,6 +72,49 @@ definition {
...
+		task ("And multiple toast alerts") {
+			Click(
+				key_text = "Success Submit",
+				locator1 = "Button#ANY");
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayAlertToast.testcase:81]: Can maybe make a macro for this. Like, `ClayAlertToast.openToastAlert`

**Comment by deborasoliveira** [comment]
ci:test:sf

**Comment by deborasoliveira** [comment]
ci:stop

**Comment by deborasoliveira** [comment]
ci:stop

**Comment by deborasoliveira** [comment]
ci:test:sf

**Comment by deborasoliveira** [comment]
ci:test:frontend

**Review by john-co — APPROVED** [review]
Looks good to me! :sunglasses:

Good job!

**Comment by john-co** [comment]
ci:forward

**Pull request closed without merging**